src/libraries/System.IO.FileSystem/tests/RandomAccess/FlushToDisk.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO.Pipes;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class RandomAccess_FlushToDisk : RandomAccess_Base<long>
+    {
+        public static IEnumerable<object[]> RandomBufferByteCounts => new[]
+        {
+            // To ensure that flushing works correctly for a variety of sizes, we want to test with
+            // buffers that are smaller than a page (e.g. just 1 byte) and buffers that are several
+            // times larger than a page (e.g. up to 10 pages).
+            new object[] { Random.Shared.Next(1, Environment.SystemPageSize * 10) },
```

Comment:
this needs to use a local Random object that was seeded. Eg., you see `new Random(42)` across many tests. Otherwise failures aren't reproducible.

Reply:
for something like the file content, it doesn't matter, as we don't think that should be relevant.

Reply:
Or instead of random it may be better to just list some fixed values eg 1, half page size, page size, etc

Reply:
Change made in [this commit](https://github.com/dotnet/runtime/pull/89100/commits/54a321514f93e2904f4ccaec28208ef8c82fddfe) 👍